### PR TITLE
Feat. request for Dash Platform: `quorum sign` rpc command with additional quorumHash #3424

### PR DESCRIPTION
**Pull request by 10xcryptodev**
add quorumHash to rpc command quorum sign as requested in #3424

### CODE REVIEW

**Comment by 10xcryptodev** [comment]
@codablock thanks for reviewing, changed to optional the parameter

File: src/rpc/rpcquorums.cpp [hunk 1]
```diff
@@ -288,12 +288,13 @@ UniValue quorum_memberof(const JSONRPCRequest& request)
 void quorum_sign_help()
 {
     throw std::runtime_error(
-            "quorum sign llmqType \"id\" \"msgHash\"\n"
+            "quorum sign llmqType \"id\" \"msgHash\" \"quorumHash\"\n"
```
Comment (nmarley) [inline src/rpc/rpcquorums.cpp:291]: I think optional params generally have parens around them here to show that it's not required.

File: src/rpc/rpcquorums.cpp [hunk 2]
```diff
@@ -360,7 +363,17 @@ UniValue quorum_sigs_cmd(const JSONRPCRequest& request)
     uint256 msgHash = ParseHashV(request.params[3], "msgHash");
 
     if (cmd == "sign") {
-        return llmq::quorumSigningManager->AsyncSignIfMember(llmqType, id, msgHash);
+        if(request.params.size() > 4){
```
Comment (nmarley) [inline src/rpc/rpcquorums.cpp:366]: Nit: I think it's clear and my personal preference is to see formatting as defined in the `.clang-format` file for new code, e.g. spaces before and after the outer parens.

File: src/rpc/rpcquorums.cpp [hunk 3]
```diff
@@ -360,7 +363,17 @@ UniValue quorum_sigs_cmd(const JSONRPCRequest& request)
...
+            uint256 quorumHash = ParseHashV(request.params[4], "quorumHash");
+            auto quorum = llmq::quorumManager->GetQuorum(llmqType, quorumHash);
+            if (!quorum) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "quorum not found");
+            }
+            llmq::quorumSigSharesManager->AsyncSign(quorum, id, msgHash);
+            return true;
+        }else{
```
Comment (nmarley) [inline src/rpc/rpcquorums.cpp:374]: Nit: Same here, this else could even be omitted and just out-dent the next line since the `if` block above it returns.

**Comment by 10xcryptodev** [comment]
@nmarley thanks for sharing about the code base format, reviewed the points

File: src/rpc/rpcquorums.cpp [hunk 4]
```diff
@@ -338,7 +339,9 @@ UniValue quorum_sigs_cmd(const JSONRPCRequest& request)
     auto cmd = request.params[0].get_str();
     if (request.fHelp || (request.params.size() != 4)) {
         if (cmd == "sign") {
-            quorum_sign_help();
+            if((request.params.size() < 4) || (request.params.size() > 5)){
```
Comment (PastaPastaPasta) [inline src/rpc/rpcquorums.cpp:342]:
```suggestion
            if ((request.params.size() < 4) || (request.params.size() > 5)) {
```

File: src/rpc/rpcquorums.cpp [hunk 5]
```diff
@@ -288,12 +288,13 @@ UniValue quorum_memberof(const JSONRPCRequest& request)
 void quorum_sign_help()
 {
     throw std::runtime_error(
-            "quorum sign llmqType \"id\" \"msgHash\"\n"
+            "quorum sign llmqType \"id\" \"msgHash\" \"( quorumHash )\"\n"
```
Comment (UdjinM6) [inline src/rpc/rpcquorums.cpp:291]:
```suggestion
            "quorum sign llmqType \"id\" \"msgHash\" ( \"quorumHash\" )\n"
```

**Comment by 10xcryptodev** [comment]
@PastaPastaPasta @UdjinM6 thanks for the fine tune, getting used  to the patterns, hope don't give you so much review work for the next PRs :)

File: src/rpc/rpcquorums.cpp [hunk 6]
```diff
@@ -360,7 +363,17 @@ UniValue quorum_sigs_cmd(const JSONRPCRequest& request)
     uint256 msgHash = ParseHashV(request.params[3], "msgHash");
 
     if (cmd == "sign") {
-        return llmq::quorumSigningManager->AsyncSignIfMember(llmqType, id, msgHash);
+        if (request.params.size() > 4) {
```
Comment (PastaPastaPasta) [inline src/rpc/rpcquorums.cpp:366]: I believe it is correct to use isNull() here
```suggestion
        if (!request.params[4].isNull()) {
```

**Comment by UdjinM6** [comment]
Ping @codablock

**Pull request merged**